4_basic_assembly/2_bitwise_operations/2_practical_bit_games/answers/3_write_code/3.asm

```diff
@@ -0,0 +1,87 @@
+; Basic Assembly
+; ==============
+;
+; 3.  Sum of distances:
+;    
+;     For a binary number x, we define the "sum of 1 distances" to be the sum of
+;     distances between every two "1" bits in x's binary representation.
+;
+;     Small example (8 bits):
+; 
+;       x = {10010100}_2
+;            7  4 2
+;      
+;       The total sum of distances: (7-4) + (7-2) + (4-2) = 3 + 5 + 2 = 10
+;
+;     Create a program that takes a number x (of size 4 bytes) as input, and
+;     outputs x's "sum of 1 distances".
+;
+; Based on SO: 
+; https://stackoverflow.com/questions/32470046/calculating-sum-of-all-the-distance/32470309#32470309
+
+format PE console
+entry start
+
+include 'win32a.inc' 
+
+; ===============================================
+section '.text' code readable executable
+
+start:
+
+    call    read_hex
+    mov     ecx,32d
+    mov     edx,0
+
+; count the number of ones in a number
+sum_bits:
+    mov     ebx,eax

```

Comment:
I am not sure about this, but I think you can do without sum_bits. However, I think that it is better to first make it work with sum_bits and only then realize how to remove it.